### DISCUSSION TITLE
Render with agu-pdf in manuscript tutorial does not work

### DISCUSSION THREAD

**Discussion by StefKirsch** [quarto-dev/quarto-cli#8201] [Q&A]
### Description

I have been following [this](https://quarto.org/docs/manuscripts/authoring/rstudio.html) tutorial on creating manuscripts with Quarto in RStudio. Most of it works well, but I did not manage to get the PDF version of the paper to render properly.
Unfortunately, we still live in a day and age where producing a PDF of your manuscript is still required from researchers, so this one is pretty crucial imho!

Things I've done/tried:
- Fully updated RStudio and all packages
- Cloned the template repository [here](https://github.com/quarto-ext/manuscript-template-rstudio)
- Installed [Quarto CLI](https://quarto.org/docs/manuscripts/#install) Release candidate 1.4.538
- Rendered the manuscript to HTML (which works)
- Clicking the PDF (agu) link in the local preview 
   - This opens a render window, but after a while I get "We can't open this file. Something went wrong." (I'm using Edge)
   - There is no PDF created in my project
- Selected "Render agu-pdf" in RStudio
   - This fails with the following message (see below)
   - I quickly see a PDF file popping up in my file preview tab, but it disappears right away
- After this, I get a index.tex file in _manuscript -> _tex. I get the option in RStudio to render this to PDF, but the formatting is messed up with big vertical whitespaces. (see attachment)
[index.pdf](https://github.com/quarto-dev/quarto-cli/files/13885750/index.pdf)

Error message when rendering index.qmd with agu-pdf in RStudio
```
==> quarto preview index.qmd --to agu-pdf --no-watch-inputs --no-browse



processing file: index.qmd
                                                                                                            
output file: index.knit.md

pandoc 
  to: latex
  output-file: index.tex
  standalone: true
  shift-heading-level-by: -1
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  number-sections: true
  
metadata
  block-headings: true
  knitr:
    opts_chunk:
      echo: false
  csl: _extensions/quarto-journals/agu/american-geophysical-union.csl
  documentclass: agujournal2019
  classoptions: draft
  header-includes:
    - |
      \usepackage{url} %this package should fix any errors with URLs in refs.
      \usepackage{lineno}
      \usepackage[inline]{trackchanges} %for better track changes. finalnew option will compile document with changes incorporated.
      \usepackage{soul}
      \linenumbers
  manuscript:
    article: index.qmd
    notebooks:
      - notebook: notebooks/explore-earthquakes.qmd
      - notebook: index.qmd
        title: Article Notebook
    mecaFile: index-meca.zip
  notebook-preview-options:
    back: true
  theme: cosmo
  title-block-style: manuscript
  lightbox: auto
  title: La Palma Earthquakes
  author:
    - name: Steve Purves
      orcid: 0000-0002-0760-5497
      corresponding: true
      email: steve@curvenote.com
      roles:
        - Investigation
        - Project administration
        - Software
        - Visualization
      affiliations:
        - Curvenote
    - name: Rowan Cockett
      orcid: 0000-0002-7859-8394
      corresponding: false
      roles: []
      affiliations:
        - Curvenote
  keywords:
    - La Palma
    - Earthquakes
  abstract: |
    In September 2021, a significant jump in seismic activity on the island of La Palma (Canary Islands, Spain) signaled the start of a volcanic crisis that still continues at the time of writing. Earthquake data is continually collected and published by the Instituto Geográphico Nacional (IGN). ...
  plain-language-summary: |
    Earthquake data for the island of La Palma from the September 2021 eruption is found ...
  key-points:
    - A web scraping script was developed to pull data from the Instituto Geogràphico Nacional into a machine-readable form for analysis
    - Earthquake events on La Palma are consistent with the presence of both mantle and crustal reservoirs.
  date: last-modified
  bibliography:
    - references.bib
  citation:
    container-title: Earth and Space Science
  google-scholar: true
  clear-hidden-classes: none
  remove-hidden: all
  unroll-markdown-cells: true
  

Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
running xelatex - 2
  This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  

Output created: _manuscript\index.pdf

Watching files for changes
Browse at http://localhost:3584/web/viewer.html
  /web/viewer.htmlindex.pdf (404: Not Found)
```

**Comment by cderv** [comment]
> Output created: _manuscript\index.pdf

Can you open outside of RStudio the file `index.pdf` ? It seems there is no rendering issue so the file should be opened. 


> Browse at http://localhost:3584/web/viewer.html
  /web/viewer.htmlindex.pdf (404: Not Found)

Is this the error message you are refering too ? 
The issue seems to be with the preview opening that fails. Url is wrongly built it seems. 

I'll look into that, because it does not seem to be related to manuscript or Quarto directly. Possibly an issue with RStudio IDE  directly as quarto preview works as expected in VSCODE

**Reply by cderv** [reply]
@StefKirsch which is your RStudio IDE version ? 

I wonder if this type of preview has been fixed in latest version 🤔 

I can't reproduce with latest dailies

**Reply by StefKirsch** [reply]
@cderv thanks for the ultra-fast reply ♥

I just 
- updated R (I had a version 4.3.1, now I'm on 4.3.2)
- double checked RStudio was up to date. (See my RStudio version below).
- re-cloned the template. 
 
I am getting still getting error message. But now I *can* render the PDF from the HTML version in my browser, yay!

I did notice that `index.pdf` *does* indeed get created in `_manuscript` (I probably overlooked that earlier). 

Before this is getting too confusing, here's an overview of what works for me and what doesn't:

## What works
- Rendering to HTLML in RStudio (it auto-opens in all its glory in my browser ♥)
- Rendering to PDF in RStudio to agu-pdf
- Rendering the PDF from the local HTLM in my browser when clicking on PDF-agu

## What doesn't work for me
- When rendering to PDF in RStudio it seems to try to open `/web/viewer.htmlindex.pdf` which leads to a 404 page. So, you might get the impression that rendering the PDF failed (I think that was also my earlier mistake 😅)
- The formatting when rendering to PDF in RStudio: There are excess vertical whitespaces in the PDF (note: you actually get the same when rendering the PDF from the on the version that's [hosted](https://quarto-ext.github.io/manuscript-template-rstudio/) on the [template's repo](https://github.com/quarto-ext/manuscript-template-rstudio)).

Thanks so much for the help and developing Quarto btw!

```
RStudio 2023.12.0+369 "Ocean Storm" Release (33206f75bd14d07d84753f965eaa24756eda97b7, 2023-12-14) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.12.0+369 Chrome/116.0.5845.190 Electron/26.2.4 Safari/537.36
```
